app/core/SanitizationMiddleware.ts

```diff
@@ -0,0 +1,94 @@
+'use strict';
+
+import { JsonRpcMiddleware, JsonRpcRequest } from 'json-rpc-engine';
+import { addHexPrefix } from 'ethereumjs-util';
+
+interface TransactionParamsLike {
+  from: string;
+  to: string;
+  value: string;
+  data: string;
+  gas: string;
+  gasPrice: string;
+  nonce: string;
+  fromBlock: string;
+  toBlock: string;
+  address: string;
+  topics: string[];
+  [k: string]: string | string[];
+}
+
+// we use this to clean any custom params from the txParams
+const permittedKeys = [
+  'from',
+  'to',
+  'value',
+  'data',
+  'gas',
+  'gasPrice',
+  'nonce',
+  'fromBlock',
+  'toBlock',
+  'address',
+  'topics',
+];
+
+function cloneTxParams(txParams: TransactionParamsLike | any) {
+  const sanitized = permittedKeys.reduce(
+    (copy, permitted: keyof TransactionParamsLike) => {
+      if (permitted in txParams) {
+        if (Array.isArray(txParams[permitted])) {
+          copy[permitted] = (txParams[permitted] as string[]).map((item) =>
+            sanitize(item),
+          );
+        } else {
+          copy[permitted] = sanitize(txParams[permitted]);
+        }
+      }
+      return copy;
+    },
+    {} as TransactionParamsLike,
+  );
```

Comment:
Thoughts on dropping the `TransactionParamsLike` type altogether?  It doesn't really line up with how this middleware functions in practice because any non-string values are returned unaltered. Applying that type would seem misleading and inaccurate.

If we use `Record<string, unknown>` here we don't need to do any casting:

```suggestion
  const sanitized = permittedKeys.reduce<Record<string, unknown>>(
    (copy, permitted) => {
      if (permitted in txParams) {
        const value = txParams[permitted];
        if (Array.isArray(value)) {
          copy[permitted] = value.map((item: unknown) => sanitize(item));
        } else {
          copy[permitted] = sanitize(value);
        }
      }
      return copy;
    },
    {},
  );
```

I introduced the `value` variable because TypeScript's type inference doesn't work well with properties. Pulling that out into a variable ensured TypeScript recognized it as an array after the `isArray` check.